dask/dataframe/tests/test_groupby.py

```diff
@@ -2500,3 +2500,13 @@ def test_groupby_multi_index_with_row_operations(operation):
     ddf = dd.from_pandas(df, npartitions=3)
     actual = caller(ddf.groupby(["A", ddf["A"].eq("a1")])["B"])
     assert_eq(expected, actual)
+
+
+def test_groupby_sort_as_false_affects_within_chunk_groupby_sort():
+    df = pd.DataFrame(
+        [["2021-11-24", "B"], ["2021-11-25", "A"]], columns=["DATE", "HASH"]
+    )
+
+    expected = df.groupby("HASH", sort=False).last()
+    actual = dd.from_pandas(df, npartitions=1).groupby("HASH", sort=False).last()
```

Comment:
Does this issue only apply to single-partition DataFrames? Or should we test with multiple partitions as well?

Reply:
My understanding is it only applies to single-partition DataFrames. When there are multiple partitions, the aggregation step does a sort.